src/ILCompiler.Compiler/src/Compiler/MetadataGeneration.cs

```diff
@@ -133,6 +137,9 @@ private void Graph_NewMarkedNode(DependencyNodeCore<NodeFactory> obj)
                     return;
                 }
 
+                if (method.HasInstantiation && !method.IsAbstract)
+                    ExactMethodInstantiations.AddExactMethodInstantiationEntry(_nodeFactory, method);
```

Comment:
I already see a _methodsGenerated hashset. Are you suggesting to expose that set in a public getter on MetadataGeneration, and have for instance the ExactMethodInstantiationsNode query that getter when its GetData API is invoked, instead of individual calls to AddExactMethodInstantiationEntry?